lib/Froxlor/Cron/System/MailboxsizeCron.php

```diff
@@ -38,16 +38,24 @@ public static function run()
 			$_maildir = \Froxlor\FileDir::makeCorrectDir($maildir['maildirpath']);
 
 			if (file_exists($_maildir) && is_dir($_maildir)) {
-				// 1. Check if maildirsize exists (usually when quota is enabled)
-				$maildirsize = $_maildir.'maildirsize';
-				if (file_exists($maildirsize)) {
+				$maildirsize = \Froxlor\FileDir::makeCorrectFile($_maildir . '/maildirsize');
+				
+				// When quota is enabled and maildirsize file exists, use that to calculate size
+				if (\Froxlor\Settings::Get('system.mail_quota_enabled') == 1 && file_exists($maildirsize)) {
 					\Froxlor\FroxlorLogger::getInstanceOf()->logAction(\Froxlor\FroxlorLogger::CRON_ACTION, LOG_NOTICE, 'found maildirsize file in ' . $_maildir);
 					$file = file($maildirsize, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
-					if (!empty($file[1])) {
-						$emailusage = floatval(explode(' ', $file[1])[0]);
+					// Remove header
+					array_shift($file);
+					$emailusage = 0;
+					// Sum up all the changes (line 2 -> end)
```

Comment:
the dovecot docs dont say how they store the sizes there, only about the header. You say "sum up changes", if the file only contains changes to the size, we won't get the total usage; if I read that correctly from http://www.courier-mta.org/imap/README.maildirquota.html it should be the complete size, right?

Reply:
My comment is a bit misleading, sorry. Line 1 contains the quota limit, line 2 the start value, line 3-X the changes to that start value.

Each new/removed mail will just add a new line to the file. This saves MTAs from the hassle of recalculating the mailbox size every time a user makes a change. As mentioned on your linked page, after the file size reaches 5120 bytes (due to adding new lines all the time), the MTA regenerates it, so you don't have to walk through thousands of calculations at some point. So the maildirsize always has at least two lines.

This is an example of a maildirsize file after some usage:

```
1073741824S
324923 155
1901 1
1901 1
1901 1
-39790 -20
```

That mailbox (max size: 1GB) starts with 155 mails sized 317KB, receives three new mails with ~2KB each, before I delete 20 mails. At the end, the mailbox has 138 mails with a total size of 284KB

Reply:
Got ya, neat :)